packages/xo-server/src/xapi-stats.js

```diff
@@ -384,12 +389,23 @@ export default class XapiStats {
               ? metric.getPath(testResult)
               : [findKey(metrics, metric)]
 
-          const metricValues = getValuesFromDepth(this._statsByObject, [
+          const xoObjStats = getValuesFromDepth(
+            this._statsByObject,
             uuid,
-            step,
-            'stats',
-            ...path,
-          ])
+            type === 'vm'
+              ? {
+                  hostUuid,
+                }
+              : {}
+          )
+
+          const stepStats = getValuesFromDepth(xoObjStats, String(step), {
+            endTimestamp: json.meta.end,
+            localTimestamp: localTimestamp,
+            interval: step,
+          })
```

Comment:
I think it would be best to use `getValuesFromDepth` until the `n - 1` step and doing the last one manually if necessary.